tsl/src/continuous_aggs/create.c

```diff
@@ -722,6 +726,39 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 						 errmsg(
 							 "time bucket function must reference a hypertable dimension column")));
 
+			if (list_length(fe->args) == 3)
+			{
+				/*
+				 * Currently the third argument of the bucketing function can be
+				 * only a timezone. Only immutable expressions can be specified.
+				 */
+				tz_arg = eval_const_expressions(NULL, lthird(fe->args));
+				if (IsA(tz_arg, Const))
+				{
+					Const *tz = castNode(Const, tz_arg);
+					Assert(tz->consttype == TEXTOID);
+					const char *orig_tz_name = TextDatumGetCString(tz->constvalue);
+					const char *tz_name = ts_validate_and_normalize_timezone_name(orig_tz_name);
+
+					if (!tz_name)
+					{
+						ereport(ERROR,
+								(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+								 errmsg("invalid timezone name"),
+								 errhint("%s", orig_tz_name)));
```

Comment:
I hint should suggest a way to fix/avoid the error. Prefer just folding the timezone name into the error.

```suggestion
								 errmsg("invalid timezone name \"%s\"", orig_tz_name)));
```